release/nightly_tests/dataset/read_parquet_benchmark.py

```diff
@@ -0,0 +1,85 @@
+import pyarrow as pa
+
+import ray
+from ray.data.dataset import Dataset
+
+from benchmark import Benchmark
+from parquet_data_generator import generate_data
+
+import shutil
+import tempfile
+
+
+def read_parquet(
+    root: str,
+    parallelism: int = -1,
+    use_threads: bool = False,
+    filter=None,
+    columns=None,
+) -> Dataset:
+    return ray.data.read_parquet(
+        paths=root,
+        parallelism=parallelism,
+        use_threads=use_threads,
+        filter=filter,
+        columns=columns,
+    ).fully_executed()
+
+
+def run_read_parquet_benchmark(benchmark: Benchmark):
+    # Test with different parallelism (multi-processing for single node) and threading.
+    for parallelism in [1, 2, 4]:
+        for use_threads in [True, False]:
+            test_name = f"read-parquet-downsampled-nyc-taxi-2009-{parallelism}-{use_threads}"  # noqa: E501
+            benchmark.run(
+                test_name,
+                read_parquet,
+                root="s3://anonymous@air-example-data/ursa-labs-taxi-data/downsampled_2009_full_year_data.parquet",  # noqa: E501
+                parallelism=parallelism,
+                use_threads=use_threads,
+            )
+
+    # Test with projection and filter pushdowns.
+    # Since we have projection and filter pushdown, we can run the read on the full
+    # size of one year data fast enough on a single node.
+    test_name = "read-parquet-nyc-taxi-2018-pushdown"
+    filter_expr = (pa.dataset.field("passenger_count") <= 10) & (
+        pa.dataset.field("passenger_count") > 0
+    )
+    benchmark.run(
+        test_name,
+        read_parquet,
+        root="s3://anonymous@air-example-data/ursa-labs-taxi-data/by_year/2018",
+        columns=["passenger_count", "trip_distance"],
+        filter=filter_expr,
+    )
+
+    # Test with different number files to handle: from a few to many.
+    data_dirs = []
+    for num_files in [8, 128, 1024]:
+        data_dirs.append(tempfile.mkdtemp())
+        generate_data(
+            num_rows=1000 * num_files,
```

Comment:
each file only has 1k rows. This looks like the scale is a little bit too small. Can we at least have one test, where 1 file has 1024*1024 rows?

Reply:
The focus here is num of files rather than the scale of rows, but raised it up.